permissions/helpers.py

```diff
@@ -72,14 +73,13 @@ def can_manage_service_sector_roles(user: User, service_sector: ServiceSector) -
     )
 
 
-def can_manage_unit_roles(user: User, unit: Unit) -> bool:
+def can_manage_unit_roles(user: User, units: List[Unit]) -> bool:
     permission = "can_manage_unit_roles"
+    service_sectors = ServiceSector.objects.filter(units__in=units)
     return (
         is_superuser(user)
-        or has_service_sector_permission(
-            user, list(unit.service_sectors.all()), permission
-        )
-        or has_unit_permission(user, [unit], permission)
+        or has_service_sector_permission(user, list(service_sectors.all()), permission)
```

Comment:
This does not exactly concern this Pull request, but `has_service_sector` helper function can be optimized to perform 1 query less by having it accept a queryset for the second parameter instead of a list.

For example, instead of `list(service_sectors.all())` this would be `service_sectors.all().values_list("id", flat=True)`.

For cases where a single service sector is given, it would become `ServiceSector.objects.filter(id=service_sector.id).values_list("id", flat=True)`.

Don't know how many queries this will save, but piling up multiple calls to these permissions, I think it makes a difference.

Reply:
Good observation :mag:  Since this is not in scope and we've not yet encountered any performance issues, I would leave it as is. Also to notice that we have not yet been developing the permission edit UI nor the API (there is stub api existing for that but should be developed to gql side as new api)